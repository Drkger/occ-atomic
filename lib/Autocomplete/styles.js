import spacing from '../subatomic/spacing';

export default {
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: `calc(100% - ${spacing.xTiny}px)`,
        left: 0,
        width: '100%'
    }
};
