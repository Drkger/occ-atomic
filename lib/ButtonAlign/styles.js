export default {
    sm: {
        marginTop:'19px'
    },
    md: {
        marginTop:'9px'
    },
    lg: {
        marginTop:'3px'
    }
};