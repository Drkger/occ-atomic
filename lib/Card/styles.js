import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import shadows from '../subatomic/shadows';

const styles = {
    card: {
        background:colors.white,
        borderRadius:'5px',
        padding:'30px 20px'
    },
    cardNoPadding: {
        extend: 'card',
        padding:'0'
    }
};

const pushShadows = function() {
    for (let i = 0; i<7; i++) {
        styles[`shadow${i}`] = {
            boxShadow: shadows[`lvl${i}`]
        };
    }
};

pushShadows();

export default styles;