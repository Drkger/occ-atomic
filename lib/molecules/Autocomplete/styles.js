export default {
    autoComplete: {
        position:'relative'
    },
    droplist: {
        position:'absolute',
        zIndex:'2',
        top:'49px',
        left:'0',
        width:'100%'
    },
    labels: {
        marginTop:'-18px',
        width:'100%'
    }
};