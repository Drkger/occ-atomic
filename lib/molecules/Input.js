import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';

import Icon from '../elements/Icon';

import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import icons from '../subatomic/icons';

const styles = {
    container: {
        marginBottom:'16px',
        paddingBottom:'11px',
        position:'relative'
    },
    default: {
        color:colors.grey4,
        '& $input': {
            borderColor:colors.grey1,
        },
        '& $placeholder': {
            top:'20px',
            fontSize:'16px',
            color:colors.grey6
        }
    },
    disabled: {
        color:colors.grey1,
        '& $input': {
            borderColor:colors.grey1,
            color:colors.grey1
        },
        '& $placeholder': {
            top:'20px',
            fontSize:'16px',
            color:colors.grey2
        },
        '& $pushHolder': {
            left:'0'
        }
    },
    focus: {
        color:colors.blue,
        '& $inputWrap:after': {
            width:'100%',
            left:'0'
        },
        '& $placeholder': {
            color:colors.blue
        },
        '& $pushHolder': {
            left:'0'
        }
    },
    complete: {
        color:colors.black2,
        '& $input': {
            borderColor:colors.black2,
        },
        '& $placeholder': {
            color:colors.black2
        },
        '& $pushHolder': {
            left:'0'
        }
    },
    correct: {
        color:colors.green,
        '& $input': {
            borderColor:colors.green,
        },
        '& $placeholder': {
            color:colors.green
        },
        '& $pushHolder': {
            left:'0'
        }
    },
    error: {
        color:colors.red,
        '& $input': {
            borderColor:colors.red,
        },
        '& $placeholder': {
            color:colors.red
        },
        '& $pushHolder': {
            left:'0'
        }
    },
    label: {
        boxSizing:'border-box',
        float:'left',
        display:'inline-block',
        maxWidth:'100%',
        fontFamily:fonts.body,
        fontSize:'11px',
        fontWeight:'normal',
        marginBottom:'0',
        transition:'0.3s all'
    },
    left: {
        float:'left'
    },
    push: {
        paddingTop:'15px'
    },
    counter: {
        position:'absolute',
        top:'0',
        right:'0'
    },
    placeholder: {
        position:'absolute',
        pointerEvents:'none',
        fontSize:'11px',
        top:'0'
    },
    pushHolder: {
        left:'24px'
    },
    inputWrap: {
        boxSizing:'border-box',
        position:'relative',
        '&:after': {
            content:'""',
            display:'block',
            position:'absolute',
            background:colors.blue,
            transition:'0.3s all',
            bottom:'0',
            left:'50%',
            width:'0',
            height:'1px'
        },
    },
    input: {
        boxSizing:'border-box',
        display:'flex',
        alignItems:'center',
        width:'100%',
        height:'34px',
        color:colors.black,
        fontFamily:fonts.body,
        fontWeight:'normal',
        fontSize:'16px',
        lineHeight:'16px',
        background:'transparent',
        border:'0',
        borderBottom:`1px solid ${colors.grey1}`,
        padding:'10px 0',
        marginBottom:'4px',
        outline:'none',
        transition:'0.3s all',
        '&::placeholder': {
            color:colors.grey6
        }
    },
    textarea: {
        padding:'5px 0',
        lineHeight:'22px'
    },
    select: {
        padding:'0',
        appearance:'none',
        background:icons.base(icons.arrowDown.icon(['#4E6273'])),
        backgroundSize:'24px 24px',
        backgroundPosition:'calc(100% - 5px) 5px'
    },
    expanded: {
        height:'80px'
    },
    hasIcon: {
        paddingLeft:'24px'
    },
    icon: {
        position:'absolute',
        left:'0',
        top:'19px'
    },
    hasClear: {
        paddingRight:'24px'
    },
    clear: {
        background:'none',
        border:'0',
        padding:'0',
        margin:'0',
        position:'absolute',
        right:'0',
        top:'20px',
        outline:'0',
        cursor:'pointer'
    }
};

class Input extends React.Component {
    
    constructor(props) {
        super(props);
        this.state = {
            status: 'default',
            value: props.valueProp ? props.valueProp : '',
            touched: false,
            initialValue: props.valueProp ? props.valueProp : ''
        };
        this.onFocus = this.onFocus.bind(this);
        this.onBlur = this.onBlur.bind(this);
        this.onChange = this.onChange.bind(this);
        this.onClear = this.onClear.bind(this);
        this.setAsComplete = this.setAsComplete.bind(this);
    }
    
    componentWillMount() {
        let { input, onRef } = this.props;
        let { value } = this.state;
        if (input && input.value)
            this.setState({ value: input.value });
            if (onRef) onRef(this);
    }
    
    componentDidMount() {
        let { value } = this.state;
        if (value) {
            this.setAsComplete();
        }
    }
    
    componentWillReceiveProps(props) {
        let { input, valueProp } = props;
        let { value, initialValue } = this.state;
        if (input && input.value)
            this.setState({ value: input.value });
        else if (valueProp && valueProp != initialValue)
            this.setState({ value: valueProp });
    }
    
    componentWillUnmount() {
        let { onRef } = this.props;
        if (onRef) onRef(undefined);
    }
    
    setAsComplete() {
        this.setState({ status: 'complete' });
    }
    
    onFocus({ target }) {
        const { onFocus, selectOnFocus } = this.props;
        if (selectOnFocus) target.select();
        this.setState({ status:'focus', touched: true });
        if (onFocus) onFocus();
    }
    
    onBlur({ target: {value} }) {
        const { input, onBlur } = this.props;
        this.setState({ value:value, status: value.length ? 'complete' : 'default' });
        if (onBlur) onBlur(value);
        if (input) input.onBlur(value);
    }
    
    onChange({ target: {value} }) {
        const { maxLength, input, onChange } = this.props;
        this.setState({ value:value });
        if (onChange) onChange(value);
        if (input) input.onBlur(value);
    }
    
    onClear() {
        const { input, onClear } = this.props;
        this.setState({ value:'', status: 'default' });
        if (onClear) onClear();
        if (input) input.onBlur(value);
    }
    
    setIconColor(status) {
        let iconColors = {
            default: colors.black,
            focus: colors.blue,
            complete: colors.black,
            correct: colors.green,
            error: colors.red,
            disabled: colors.grey1
        };
        return iconColors[status];
    }
    
    render() {
        const { status, value, touched } = this.state;
        const { input, classes, className, label, placeholder, counter, maxLength, type, options, name, id, style, disabled, meta, autoFocus, error, valid, clear, selectOnFocus, iconName } = this.props;
        let realStatus = status;
        let InputType = (type == "textarea" ? "textarea" : type == "select" ? "select" : "input");
        if (status != 'focus') {
            if (disabled)
                realStatus = 'disabled';
            else if ((meta && meta.error && meta.touched) || touched && error)
                realStatus = 'error';
            else if ((meta && meta.valid) || (touched && valid))
                realStatus = 'correct';
            else if (input && input.value)
                realStatus = 'complete';
        }
        let iconColor;
        if (iconName)
            iconColor = this.setIconColor(realStatus);
        if (disabled) {
            return (
                <div className={`${classes.container} ${classes.disabled} ${(!label && placeholder) && classes.push}${className ? ` ${className}` : ''}`} style={style}>
                    {label && (
                        <label className={`${classes.label} ${classes.left}`}>
                            {label}
                        </label>
                    )}
                    <label className={`${classes.input}${iconName ? ` ${classes.hasIcon}` : ''}`}>
                        {(input && input.value) ? input.value : value ? value : placeholder}
                    </label>
                </div>
            );
        }
        return (
            <div className={`${classes.container} ${classes[realStatus]} ${(!label && placeholder) && classes.push}${className ? ` ${className}` : ''}`} style={style}>
                {label && (
                    <label
                        className={`${classes.label} ${classes.left}`}>
                        {label}
                    </label>
                )}
                {(!label && placeholder) && (
                    <label
                        className={`${classes.label} ${classes.left} ${classes.placeholder}${iconName ? ` ${classes.pushHolder}` : ''}`}>
                        {placeholder}
                    </label>
                )}
                {iconName && (
                    <Icon iconName={iconName} width={24} height={24} display="inline-block" className={classes.icon} colors={[iconColor]} />
                )}
                {(counter && status == 'focus') && (
                    <label
                        className={`${classes.label} ${classes.right} ${classes.counter}`}>
                        {maxLength-value.length}
                    </label>
                )}
                <div className={classes.inputWrap}>
                    {type == "select" ? (
                        <InputType
                            type={type}
                            name={name}
                            id={id}
                            className={`${classes.input} ${classes.select}${iconName ? ` ${classes.hasIcon}` : ''}${clear ? ` ${classes.hasClear}` : ''}${type == "textarea" ? ` ${classes.textarea}` : ''}${(value && type=="textarea") ? ` ${classes.expanded}` : ''}`}
                            placeholder={(label && placeholder) ? placeholder : ''}
                            value={(input && input.value) ? input.value : value}
                            autoFocus={autoFocus}
                            maxLength={maxLength}
                            onFocus={this.onFocus}
                            onBlur={this.onBlur}
                            onChange={this.onChange}
                            ref = {ref => { this.child = ref; }}
                        >
                            {options.map(option => (
                                <option key={option.value} value={option.value}>{option.text}</option>
                            ))}
                        </InputType>
                    ) : (
                        <InputType
                            type={type}
                            name={name}
                            id={id}
                            className={`${classes.input}${iconName ? ` ${classes.hasIcon}` : ''}${clear ? ` ${classes.hasClear}` : ''}${type == "textarea" ? ` ${classes.textarea}` : ''}${(value && type=="textarea") ? ` ${classes.expanded}` : ''}`}
                            placeholder={(label && placeholder) ? placeholder : ''}
                            value={(input && input.value) ? input.value : value}
                            autoFocus={autoFocus}
                            maxLength={maxLength}
                            onFocus={this.onFocus}
                            onBlur={this.onBlur}
                            onChange={this.onChange}
                            ref = {ref => { this.child = ref; }}
                        />
                    )}
                </div>
                {(value && clear) && (
                    <button onClick={this.onClear} className={classes.clear}><Icon iconName="close" /></button>
                )}
                {((meta && meta.error && meta.touched) || (error && touched) && status != 'focus') && (
                    <label className={`${classes.label} ${classes.error}`}>{meta ? meta.error : error}</label>
                )}
            </div>
        );
    }
}

Input.defaultProps = {
    type: 'text',
    counter: false,
    maxLength: 100,
    disabled: false,
    autoFocus: false,
    clear: false,
    valueProp: '',
    selectOnFocus: false
};

Input.propTypes = {
    classes: PropTypes.object,
    className: PropTypes.string,
    type: PropTypes.string,
    maxLength: PropTypes.number,
    label: PropTypes.string,
    placeholder: PropTypes.string,
    name: PropTypes.string,
    id: PropTypes.string,
    style: PropTypes.object,
    disabled: PropTypes.bool,
    counter: PropTypes.bool,
    autoFocus: PropTypes.bool,
    clear: PropTypes.bool,
    error: PropTypes.string,
    valid: PropTypes.bool,
    valueProp: PropTypes.node,
    selectOnFocus: PropTypes.bool,
    meta: PropTypes.object,
    input: PropTypes.object,
    onFocus: PropTypes.func,
    onBlur: PropTypes.func,
    onChange: PropTypes.func,
    onClear: PropTypes.func,
    onRef: PropTypes.func,
    options: PropTypes.array,
    iconName: PropTypes.string
};

export default injectSheet(styles)(Input);