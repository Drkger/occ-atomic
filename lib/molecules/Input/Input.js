import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';

import Icon from '../../elements/Icon';

import colors from '../../subatomic/colors';

class Input extends React.Component {
    
    constructor(props) {
        super(props);
        this.state = {
            status: 'default',
            value: props.valueProp ? props.valueProp : '',
            touched: false,
            initialValue: props.valueProp ? props.valueProp : ''
        };
        this.onFocus = this.onFocus.bind(this);
        this.onBlur = this.onBlur.bind(this);
        this.onChange = this.onChange.bind(this);
        this.onClear = this.onClear.bind(this);
        this.setAsComplete = this.setAsComplete.bind(this);
    }
    
    componentWillMount() {
        let { input, onRef } = this.props;
        let { value } = this.state;
        if (input && input.value)
            this.setState({ value: input.value });
            if (onRef) onRef(this);
    }
    
    componentDidMount() {
        let { value } = this.state;
        if (value) {
            this.setAsComplete();
        }
    }
    
    componentWillReceiveProps(props) {
        let { input, valueProp } = props;
        let { value, initialValue } = this.state;
        if (input && input.value)
            this.setState({ value: input.value });
        else if (valueProp && valueProp != initialValue)
            this.setState({ value: valueProp });
    }
    
    componentWillUnmount() {
        let { onRef } = this.props;
        if (onRef) onRef(undefined);
    }
    
    setAsComplete() {
        this.setState({ status: 'complete' });
    }
    
    onFocus({ target }) {
        const { onFocus, selectOnFocus } = this.props;
        if (selectOnFocus) target.select();
        this.setState({ status:'focus', touched: true });
        if (onFocus) onFocus();
    }
    
    onBlur({ target: {value} }) {
        const { input, onBlur } = this.props;
        this.setState({ value:value, status: value.length ? 'complete' : 'default' });
        if (onBlur) onBlur(value);
        if (input) input.onBlur(value);
    }
    
    onChange({ target: {value} }) {
        const { maxLength, input, onChange } = this.props;
        this.setState({ value:value });
        if (onChange) onChange(value);
        if (input) input.onBlur(value);
    }
    
    onClear() {
        const { input, onClear } = this.props;
        this.setState({ value:'', status: 'default' });
        if (onClear) onClear();
        if (input) input.onBlur(value);
    }
    
    setIconColor(status) {
        let iconColors = {
            default: colors.black,
            focus: colors.blue,
            complete: colors.black,
            correct: colors.green,
            error: colors.red,
            disabled: colors.grey1
        };
        return iconColors[status];
    }
    
    render() {
        const { status, value, touched } = this.state;
        const { input, classes, className, label, placeholder, counter, maxLength, type, options, name, id, style, disabled, meta, autoFocus, error, valid, clear, selectOnFocus, iconName } = this.props;
        let realStatus = status;
        let InputType = (type == "textarea" ? "textarea" : type == "select" ? "select" : "input");
        if (status != 'focus') {
            if (disabled)
                realStatus = 'disabled';
            else if ((meta && meta.error && meta.touched) || touched && error)
                realStatus = 'error';
            else if ((meta && meta.valid) || (touched && valid))
                realStatus = 'correct';
            else if (input && input.value)
                realStatus = 'complete';
        }
        let iconColor;
        if (iconName)
            iconColor = this.setIconColor(realStatus);
        if (disabled) {
            return (
                <div className={`${classes.container} ${classes.disabled} ${(!label && placeholder) && classes.push}${className ? ` ${className}` : ''}`} style={style}>
                    {label && (
                        <label className={`${classes.label} ${classes.left}`}>
                            {label}
                        </label>
                    )}
                    <label className={`${classes.input}${iconName ? ` ${classes.hasIcon}` : ''}`}>
                        {(input && input.value) ? input.value : value ? value : placeholder}
                    </label>
                </div>
            );
        }
        return (
            <div className={`${classes.container} ${classes[realStatus]} ${(!label && placeholder) && classes.push}${className ? ` ${className}` : ''}`} style={style}>
                {label && (
                    <label
                        className={`${classes.label} ${classes.left}`}>
                        {label}
                    </label>
                )}
                {(!label && placeholder) && (
                    <label
                        className={`${classes.label} ${classes.left} ${classes.placeholder}${iconName ? ` ${classes.pushHolder}` : ''}`}>
                        {placeholder}
                    </label>
                )}
                {iconName && (
                    <Icon iconName={iconName} width={24} height={24} display="inline-block" className={classes.icon} colors={[iconColor]} />
                )}
                {(counter && status == 'focus') && (
                    <label
                        className={`${classes.label} ${classes.right} ${classes.counter}`}>
                        {maxLength-value.length}
                    </label>
                )}
                <div className={classes.inputWrap}>
                    {type == "select" ? (
                        <InputType
                            type={type}
                            name={name}
                            id={id}
                            className={`${classes.input} ${classes.select}${iconName ? ` ${classes.hasIcon}` : ''}${clear ? ` ${classes.hasClear}` : ''}${type == "textarea" ? ` ${classes.textarea}` : ''}${(value && type=="textarea") ? ` ${classes.expanded}` : ''}`}
                            placeholder={(label && placeholder) ? placeholder : ''}
                            value={(input && input.value) ? input.value : value}
                            autoFocus={autoFocus}
                            maxLength={maxLength}
                            onFocus={this.onFocus}
                            onBlur={this.onBlur}
                            onChange={this.onChange}
                            ref = {ref => { this.child = ref; }}
                        >
                            {options.map(option => (
                                <option key={option.value} value={option.value}>{option.text}</option>
                            ))}
                        </InputType>
                    ) : (
                        <InputType
                            type={type}
                            name={name}
                            id={id}
                            className={`${classes.input}${iconName ? ` ${classes.hasIcon}` : ''}${clear ? ` ${classes.hasClear}` : ''}${type == "textarea" ? ` ${classes.textarea}` : ''}${(value && type=="textarea") ? ` ${classes.expanded}` : ''}`}
                            placeholder={(label && placeholder) ? placeholder : ''}
                            value={(input && input.value) ? input.value : value}
                            autoFocus={autoFocus}
                            maxLength={maxLength}
                            onFocus={this.onFocus}
                            onBlur={this.onBlur}
                            onChange={this.onChange}
                            ref = {ref => { this.child = ref; }}
                        />
                    )}
                </div>
                {(value && clear) && (
                    <button onClick={this.onClear} className={classes.clear}><Icon iconName="close" /></button>
                )}
                {((meta && meta.error && meta.touched) || (error && touched) && status != 'focus') && (
                    <label className={`${classes.label} ${classes.error}`}>{meta ? meta.error : error}</label>
                )}
            </div>
        );
    }
}

Input.defaultProps = {
    type: 'text',
    counter: false,
    maxLength: 100,
    disabled: false,
    autoFocus: false,
    clear: false,
    valueProp: '',
    selectOnFocus: false
};

Input.propTypes = {
    classes: PropTypes.object,
    className: PropTypes.string,
    type: PropTypes.string,
    maxLength: PropTypes.number,
    label: PropTypes.string,
    placeholder: PropTypes.string,
    name: PropTypes.string,
    id: PropTypes.string,
    style: PropTypes.object,
    disabled: PropTypes.bool,
    counter: PropTypes.bool,
    autoFocus: PropTypes.bool,
    clear: PropTypes.bool,
    error: PropTypes.string,
    valid: PropTypes.bool,
    valueProp: PropTypes.node,
    selectOnFocus: PropTypes.bool,
    meta: PropTypes.object,
    input: PropTypes.object,
    onFocus: PropTypes.func,
    onBlur: PropTypes.func,
    onChange: PropTypes.func,
    onClear: PropTypes.func,
    onRef: PropTypes.func,
    options: PropTypes.array,
    iconName: PropTypes.string
};

export default Input;