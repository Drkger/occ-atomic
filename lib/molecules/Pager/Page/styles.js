import colors from '../../../subatomic/colors';

export default {
    li: {
        display:'inline-block',
        cursor:'pointer',
        padding:'2px 11px',
        marginLeft:'4px',
        marginRight:'4px',
        borderRadius:'13px',
        cursor:'pointer',
        transition:'0.3s all',
        outline:'0',
        '&:hover': {
            background:'#ECECEE'
        }
    },
    active: {
        background:`${colors.blue} !important`,
        color:colors.white
    }
};