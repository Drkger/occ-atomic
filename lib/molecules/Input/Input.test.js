import React from 'react';
import { shallow } from 'enzyme';
import Input from './Input';
import styles from './styles';
import InputJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("Input", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <Input classes={classes} />
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("InputJSS", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <InputJSS />
        );
        expect(wrapper).toMatchSnapshot();
    });

});