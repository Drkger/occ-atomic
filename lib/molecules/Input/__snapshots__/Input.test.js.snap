// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    autoFocus={false}
    classes={
      Object {
        "clear": "clear",
        "complete": "complete",
        "container": "container",
        "correct": "correct",
        "counter": "counter",
        "default": "default",
        "disabled": "disabled",
        "error": "error",
        "expanded": "expanded",
        "focus": "focus",
        "hasClear": "hasClear",
        "hasIcon": "hasIcon",
        "icon": "icon",
        "input": "input",
        "inputWrap": "inputWrap",
        "label": "label",
        "left": "left",
        "placeholder": "placeholder",
        "push": "push",
        "pushHolder": "pushHolder",
        "select": "select",
        "textarea": "textarea",
      }
    }
    clear={false}
    counter={false}
    disabled={false}
    maxLength={100}
    selectOnFocus={false}
    type="text"
    valueProp=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        undefined,
        undefined,
        false,
        <div
          className="inputWrap"
        >
          <input
            autoFocus={false}
            className="input"
            id={undefined}
            maxLength={100}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder=""
            type="text"
            value=""
          />
        </div>,
        "",
        undefined,
      ],
      "className": "container default undefined",
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      undefined,
      undefined,
      undefined,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            autoFocus={false}
            className="input"
            id={undefined}
            maxLength={100}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder=""
            type="text"
            value=""
          />,
          "className": "inputWrap",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoFocus": false,
            "className": "input",
            "id": undefined,
            "maxLength": 100,
            "name": undefined,
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "placeholder": "",
            "type": "text",
            "value": "",
          },
          "ref": [Function],
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      "",
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          undefined,
          undefined,
          false,
          <div
            className="inputWrap"
          >
            <input
              autoFocus={false}
              className="input"
              id={undefined}
              maxLength={100}
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder=""
              type="text"
              value=""
            />
          </div>,
          "",
          undefined,
        ],
        "className": "container default undefined",
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        undefined,
        undefined,
        undefined,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              autoFocus={false}
              className="input"
              id={undefined}
              maxLength={100}
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder=""
              type="text"
              value=""
            />,
            "className": "inputWrap",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoFocus": false,
              "className": "input",
              "id": undefined,
              "maxLength": 100,
              "name": undefined,
              "onBlur": [Function],
              "onChange": [Function],
              "onFocus": [Function],
              "placeholder": "",
              "type": "text",
              "value": "",
            },
            "ref": [Function],
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        "",
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`InputJSS matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Input)
    autoFocus={false}
    clear={false}
    counter={false}
    disabled={false}
    maxLength={100}
    selectOnFocus={false}
    type="text"
    valueProp=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoFocus": false,
      "classes": Object {
        "clear": "Input-clear-0-1-22",
        "complete": "Input-complete-0-1-5",
        "container": "Input-container-0-1-1",
        "correct": "Input-correct-0-1-6",
        "counter": "Input-counter-0-1-11",
        "default": "Input-default-0-1-2",
        "disabled": "Input-disabled-0-1-3",
        "error": "Input-error-0-1-7",
        "expanded": "Input-expanded-0-1-18",
        "focus": "Input-focus-0-1-4",
        "hasClear": "Input-hasClear-0-1-21",
        "hasIcon": "Input-hasIcon-0-1-19",
        "icon": "Input-icon-0-1-20",
        "input": "Input-input-0-1-15",
        "inputWrap": "Input-inputWrap-0-1-14",
        "label": "Input-label-0-1-8",
        "left": "Input-left-0-1-9",
        "placeholder": "Input-placeholder-0-1-12",
        "push": "Input-push-0-1-10",
        "pushHolder": "Input-pushHolder-0-1-13",
        "select": "Input-select-0-1-17",
        "textarea": "Input-textarea-0-1-16",
      },
      "clear": false,
      "counter": false,
      "disabled": false,
      "maxLength": 100,
      "selectOnFocus": false,
      "type": "text",
      "valueProp": "",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoFocus": false,
        "classes": Object {
          "clear": "Input-clear-0-1-22",
          "complete": "Input-complete-0-1-5",
          "container": "Input-container-0-1-1",
          "correct": "Input-correct-0-1-6",
          "counter": "Input-counter-0-1-11",
          "default": "Input-default-0-1-2",
          "disabled": "Input-disabled-0-1-3",
          "error": "Input-error-0-1-7",
          "expanded": "Input-expanded-0-1-18",
          "focus": "Input-focus-0-1-4",
          "hasClear": "Input-hasClear-0-1-21",
          "hasIcon": "Input-hasIcon-0-1-19",
          "icon": "Input-icon-0-1-20",
          "input": "Input-input-0-1-15",
          "inputWrap": "Input-inputWrap-0-1-14",
          "label": "Input-label-0-1-8",
          "left": "Input-left-0-1-9",
          "placeholder": "Input-placeholder-0-1-12",
          "push": "Input-push-0-1-10",
          "pushHolder": "Input-pushHolder-0-1-13",
          "select": "Input-select-0-1-17",
          "textarea": "Input-textarea-0-1-16",
        },
        "clear": false,
        "counter": false,
        "disabled": false,
        "maxLength": 100,
        "selectOnFocus": false,
        "type": "text",
        "valueProp": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
